assets/recipient-form.js

```diff
@@ -1,142 +1,164 @@
 if (!customElements.get('recipient-form')) {
-  customElements.define(
-    'recipient-form',
-    class RecipientForm extends HTMLElement {
-      constructor() {
-        super();
-        this.checkboxInput = this.querySelector(`#Recipient-Checkbox-${this.dataset.sectionId}`);
-        this.checkboxInput.disabled = false;
-        this.hiddenControlField = this.querySelector(`#Recipient-Control-${this.dataset.sectionId}`);
-        this.hiddenControlField.disabled = true;
-        this.emailInput = this.querySelector(`#Recipient-email-${this.dataset.sectionId}`);
-        this.nameInput = this.querySelector(`#Recipient-name-${this.dataset.sectionId}`);
-        this.messageInput = this.querySelector(`#Recipient-message-${this.dataset.sectionId}`);
-        this.errorMessageWrapper = this.querySelector('.product-form__recipient-error-message-wrapper');
-        this.errorMessageList = this.errorMessageWrapper?.querySelector('ul');
-        this.errorMessage = this.errorMessageWrapper?.querySelector('.error-message');
-        this.defaultErrorHeader = this.errorMessage?.innerText;
-        this.currentProductVariantId = this.dataset.productVariantId;
-        this.addEventListener('change', this.onChange.bind(this));
-      }
-
-      cartUpdateUnsubscriber = undefined;
-      variantChangeUnsubscriber = undefined;
-      cartErrorUnsubscriber = undefined;
+  customElements.define('recipient-form', class RecipientForm extends HTMLElement {
+    constructor() {
+      super();
+      this.checkboxInput = this.querySelector(`#Recipient-Checkbox-${ this.dataset.sectionId }`);
+      this.checkboxInput.disabled = false;
+      this.hiddenControlField = this.querySelector(`#Recipient-Control-${ this.dataset.sectionId }`);
+      this.hiddenControlField.disabled = true;
+      this.emailInput = this.querySelector(`#Recipient-email-${ this.dataset.sectionId }`);
+      this.nameInput = this.querySelector(`#Recipient-name-${ this.dataset.sectionId }`);
+      this.messageInput = this.querySelector(`#Recipient-message-${ this.dataset.sectionId }`);
+      this.sendonInput = this.querySelector(`#Recipient-send_on-${ this.dataset.sectionId }`);
+      this.offsetProperty = this.querySelector(`#Recipient-Offset-${ this.dataset.sectionId }`);
+      if (this.offsetProperty) this.offsetProperty.value = new Date().getTimezoneOffset();
```

Comment:
Out of curiosity, is there any back end validation/erroring if the date entered isn't in the future?

Say I make a mistake and enter `2022` as the year.

Reply:
Yeah, we only allow from today until 90 days currently, though that is an arbitrary choice.

Reply:
Sounds good 👍